src/tests/java/com/util/CodeUtilTest.java

```diff
@@ -0,0 +1,71 @@
+package com.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CodeUtilTest {
+
+    @Test
+    public void testEscape() {
+        Assert.assertEquals("%80%u9e00Ё%00A",
+                CodeUtil.escape("\u0080\u9e00\u0401\u0000A"));
+    }
+
+    @Test
+    public void testUnescape() {
+        Assert.assertEquals("fo\u0011ar", CodeUtil.unescape("fo%11ar"));
+    }
+
+    @Test
+    public void testDbEncode() {
+        Assert.assertNull(CodeUtil.dbEncode(null));
+        Assert.assertEquals("foo", CodeUtil.dbEncode("foo"));
+    }
+
+    @Test
+    public void testDbDecode() {
+        Assert.assertNull(CodeUtil.dbDecode(null));
+        Assert.assertEquals("foo", CodeUtil.dbDecode("foo"));
+    }
+
+    @Test
+    public void testToText() {
+        Assert.assertEquals("", CodeUtil.toText(null));
+        Assert.assertEquals("foo", CodeUtil.toText("foo"));
+    }
+
+    @Test
+    public void testToHtml() {
+        Assert.assertEquals("", CodeUtil.toHtml(null));
+        Assert.assertEquals("foo", CodeUtil.toHtml("foo"));
+    }
+
+    @Test
+    public void testToUnicode() {
+        Assert.assertEquals("1",
+                CodeUtil.toUnicode("1", true));
+        Assert.assertEquals(">\\ \\\\\\u0000\\r\\t\\n\\f\\!",
+                CodeUtil.toUnicode("> \\\u0000\r\t\n\f!", true));
+    }
+
+    @Test
+    public void testFromUnicode1() {
+        CodeUtil codeUtil = new CodeUtil();
+        char[] in = {'t', '\\', '\\', 'p', '\\', 'u', '0', 'a', 'A', 'a'};
+        char[] convtBuf = {'\u0139', '\u02f9', '\u01e1', '\u001e', 'p', '@', 'U'};
+
+        Assert.assertEquals("\\pપ",
+                codeUtil.fromUnicode(in, 1, 8, convtBuf));
+    }
+
+    @Test
+    public void testFromUnicode2() {
+        CodeUtil codeUtil = new CodeUtil();
```

Comment:
I don't think there's a need to use an entirely new object & test here. You could have them as part of the 1st test.

Reply:
Done